integration/spark/spark3/src/main/java/io/openlineage/spark3/agent/lifecycle/plan/column/ExpressionDependencyCollector.java

```diff
@@ -145,6 +151,37 @@ static void collectFromNode(ColumnLevelLineageContext context, LogicalPlan node)
         });
   }
 
+  private static void extensionColumnLineage(
```

Comment:
Would it make sense move this code into separate class like `UnionDependencyVisitor` ?

Reply:
Alternatively one could implement CustomCollectorVisitor and add it into the results of `CustomCollectorsUtils.collectExpressionDependencies`